synnax/pkg/api/channel.go

```diff
@@ -262,10 +264,17 @@ func translateChannelsBackward(channels []Channel) ([]channel.Channel, error) {
 			LocalIndex:  ch.Index.LocalKey(),
 			Virtual:     ch.Virtual,
 			Internal:    ch.Internal,
+			Expression:  ch.Expression,
+			Requires:    ch.Requires,
 		}
 		if ch.IsIndex {
 			tCH.LocalIndex = tCH.LocalKey
 		}
+
+		isCalc := ch.Expression != "" && !ch.IsIndex && ch.Virtual && !ch.Internal
+		if isCalc {
+			tCH.LocalKey = ch.Key.LocalKey()
```

Comment:
This is unintuitive. Why do we only do it for calculated channels?

Reply:
I dont remember why i did this specifically for calc channels so I'm just gonna copy over the key everytime. is that a problem